### PR TITLE
Refactored based on new datamodel, added more tests.

### PR DESCRIPTION
**Pull request by AndreSoderlund**
## Types of changes

- [ ] Bug fix
- [x] New feature
- [x] Removed feature
- [x] Code style update (formatting etc.)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes

## Does this PR introduce a breaking change?
- [x] Yes (I have stepped the version number accordingly)
- [ ] No
      
## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly (if applicable).
- [x] I have added/updated tests to cover my changes (if applicable).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -43,11 +43,6 @@
 			<artifactId>hibernate-jpamodelgen</artifactId>
 			<scope>provided</scope>
 		</dependency>
-		<dependency>
-			<groupId>net.kaczmarzyk</groupId>
-			<artifactId>specification-arg-resolver</artifactId>
-			<version>${specification-arg-resolver.version}</version>
```
Comment (josc04) [inline pom.xml:49]: Remove `specification-arg-resolver.version` on row 21 also?

File: src/test/resources/db/schema.sql [hunk 2]
```diff
@@ -16,28 +16,16 @@ create table api_vuxenutb.KurserVuxenutbildning
...
-    Amneskyp varchar(4096) COLLATE Finnish_Swedish_CI_AS,
-    Amneskod varchar(4096) COLLATE Finnish_Swedish_CI_AS,
-    OmradeId int not null,
-    Omrade varchar(4096) COLLATE Finnish_Swedish_CI_AS,
-    OmradeEng varchar(4096) COLLATE Finnish_Swedish_CI_AS,
-    InriktningId int not null,
-    Inriktning varchar(4096) COLLATE Finnish_Swedish_CI_AS,
-    InriktningEng varchar(4096) COLLATE Finnish_Swedish_CI_AS
+	AntalPlatser int,
+    Undervisningssprak varchar(4096) COLLATE Finnish_Swedish_CI_AS,
```
Comment (josc04) [inline src/test/resources/db/schema.sql:27]: Check indentation 😉

Comment (Cheezi747) [reply src/test/resources/db/schema.sql:27]: Good luck changing the indentation in `schema.sql` and getting 

`Call Java CI with Maven / shared-workflows / Run CI (pull_request) Successful in 1m `

Comment (AndreSoderlund) [reply src/test/resources/db/schema.sql:27]: The indentation is correct for me locally ¯\_(ツ)_/¯

**Review by josc04 — APPROVED** [review]

**Pull request merged**